index.js

```diff
@@ -1,300 +1,147 @@
-    else if (input.match(/\.box\d*$/)) return encoder.box(input)
-    else return encoder.string(input)
-  } else if (typeof input === 'boolean') {
-    return encoder.boolean(input)
-  } else {
-    if (!Number.isInteger(input) && !Buffer.isBuffer(input))
-      console.warn('not encoding unknown value', input)
-    // FIXME: more checks, including floats!
-    return input
   }
+
+  throw new Error('cannot encoding, type is not defined')
 }
 
 const decoder = {
-  feed(buf) {
-    const tf = buf.slice(0, 2)
-    const d = buf.slice(2)
-
-    let feedExtension
-    if (tf.equals(CLASSIC_FEED_TF)) feedExtension = '.ed25519'
-    else if (tf.equals(GABBYGR_FEED_TF)) feedExtension = '.ggfeed-v1'
-    else if (tf.equals(BENDYBT_FEED_TF)) feedExtension = '.bbfeed-v1'
-    else throw new Error('Unknown feed: ' + buf)
-
-    const b64part = d.toString('base64')
-    return '@' + b64part + feedExtension
+  classic(input, type, format) {
+    return [
+      type.sigil || '',
+      input.slice(2).toString('base64'),
+      format.suffix || '',
+    ].join('')
   },
+  bool(input) {
+    if (input.size > 3) throw new Error('boolean BFE must be 3 bytes')
+    if (input.slice(2, 3).equals(BOOL.FALSE)) return false
+    if (input.slice(2, 3).equals(BOOL.TRUE)) return true
 
-  message(buf) {
-    const tf = buf.slice(0, 2)
-    const d = buf.slice(2)
-
-    let msgExtension
-    if (tf.equals(CLASSIC_MSG_TF)) msgExtension = '.sha256'
-    else if (tf.equals(GABBYGR_MSG_TF)) msgExtension = '.ggmsg-v1'
-    else if (tf.equals(BENDYBT_MSG_TF)) msgExtension = '.bbmsg-v1'
-    else throw new Error('Unknown msg: ' + buf)
-
-    const b64part = d.toString('base64')
-    return '%' + b64part + msgExtension
+    throw new Error('invalid boolean BFE')
   },
+}
 
-  blob(buf) {
-    const tf = buf.slice(0, 2)
-    const d = buf.slice(2)
-
-    let blobExtension
-    if (tf.equals(CLASSIC_BLOB_TF)) blobExtension = '.sha256'
-    else throw new Error('Unknown blob ID: ' + buf)
-
-    const b64part = d.toString('base64')
-    return '&' + b64part + blobExtension
-  },
+function decode(input) {
+  /* cases we don't decode */
+  if (input === null) return null
+  if (Number.isInteger(input)) return input
 
-  signature(buf) {
-    const d = buf.slice(2)
-    const b64part = d.toString('base64')
-    return b64part + '.sig.ed25519'
-  },
+  if (Buffer.isBuffer(input)) {
+    if (input.length < 2)
+      throw new Error('Buffer is missing first two type & format bytes')
 
-  box(buf) {
-    const tf = buf.slice(0, 2)
-    const d = buf.slice(2)
-    if (tf.equals(BOX1_TF)) return d.toString('base64') + '.box'
-    else if (tf.equals(BOX2_TF)) return d.toString('base64') + '.box2'
-    else throw new Error('Unknown box: ' + buf)
-  },
+    if (input.equals(NIL.TFD)) return null
+    if (input.slice(0, 2).equals(BOOL.TF)) return decoder.bool(input)
+    if (input.slice(0, 2).equals(STRING.TF))
+      return input.slice(2).toString('utf8')
```

Comment:
I liked how it was before: there was a decoder object with `decoder.string` which decoding details for each case. Now this is bringing them back, all concerns mixed together.

Reply:
agree. Have moved this into `decoder.string`